pkg/aws/ec2Client.go

```diff
@@ -2,116 +2,165 @@ package aws
 
 import (
 	"context"
-	"os"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/pkg/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/giantswarm/aws-network-topology-operator/pkg/k8sclient"
 )
 
 type EC2Client struct {
-	ctx               context.Context
-	ec2Client         *ec2.Client
-	k8sClient         *k8sclient.Cluster
-	managementCluster types.NamespacedName
+	ctx       context.Context
+	ec2Client *ec2.Client
+	k8sClient *k8sclient.Cluster
+	cluster   types.NamespacedName
 }
 
-func NewEC2Client(ctx context.Context, k8sClient *k8sclient.Cluster, managementCluster types.NamespacedName) *EC2Client {
+func NewEC2Client(ctx context.Context, k8sClient *k8sclient.Cluster, cluster types.NamespacedName) *EC2Client {
 	return &EC2Client{
-		ctx:               ctx,
-		ec2Client:         nil,
-		k8sClient:         k8sClient,
-		managementCluster: managementCluster,
+		ctx:       ctx,
+		ec2Client: nil,
+		k8sClient: k8sClient,
+		cluster:   cluster,
 	}
 }
 
-func (e *EC2Client) client() *ec2.Client {
+func (e *EC2Client) client() (*ec2.Client, error) {
```

Comment:
Function rewritten to turn hard `os.Exit` into returning an error, and to allow using this for workload clusters as well since had nothing specific to management clusters